javalang/parser.py

```diff
@@ -1731,7 +1740,23 @@ def parse_expressionl(self):
             return tree.TernaryExpression(condition=expression_2,
                                           if_true=true_expression,
                                           if_false=false_expression)
-
+        if self.would_accept('->'):
+            body = self.parse_lambda_method_body()
+            return tree.LambdaExpression(parameters=[expression_2],
+                                         body=body)
+        if self.try_accept('::'):
+            if self.would_accept('<'):
```

Comment:
Right so I am now handling `new` and the `super` cases. I am still not handing the following cases:
* `Primary :: [TypeArguments] Identifier`
*  `ArrayType :: new`